src/main/groovy/wooga/gradle/upm/artifactory/UPMProjectDeclaration.groovy

```diff
@@ -23,14 +20,17 @@ class UPMProjectDeclaration implements UPMPackSpec, Named {
         Extensions.setPropertiesOwner(UPMProjectDeclaration, extension, name)
         extension.with {
             version.convention(UPMArtifactoryConventions.version.resolve(name).getStringValueProvider(project).orElse(project.provider { project.version.toString() }))
-            Provider<Directory> defaultDir = UPMArtifactoryConventions.packageDirectory.resolve(name)
-                    .getDirectoryValueProvider(project, null, providers.provider{project.layout.projectDirectory})
-            packageDirectory.convention(defaultDir)
+            packageDirectory.convention(resolvePackageDirectory(project, name))
             generateMetaFiles.convention(UPMArtifactoryConventions.generateMetaFiles.resolve(name).getBooleanValueProvider(project))
         }
         return extension
     }
 
+    static final Provider<Directory> resolvePackageDirectory(Project project, String name) {
+        UPMArtifactoryConventions.packageDirectory.resolve(name)
+                .getDirectoryValueProvider(project, null, project.providers.provider{project.layout.projectDirectory})
```

Comment:
I would use brackets here and proper spacing. Otherwise it is hard to read.
```suggestion
                .getDirectoryValueProvider(project, null, project.providers.provider({ project.layout.projectDirectory }))
```

Reply:
I literally took 1 existing function, and moved it somewhere else. I'm not aware of our atlas-groovy plugins  codestyle and/or it's consistency, but am happy to do so if you point me to where it's defined / or give me examples of which parts I should take as a reference.

Reply:
Just saw it. No major blocker.